target_postgres/connector.py

```diff
@@ -180,8 +180,10 @@ def copy_table_structure(
 
     @contextmanager
     def _connect(self) -> t.Iterator[sqlalchemy.engine.Connection]:
-        with self._engine.connect().execution_options() as conn:
+        engine = self._engine
+        with engine.connect().execution_options() as conn:
             yield conn
+        engine.dispose()
```

Comment:
This spot relates to an issue we observed after adding more test cases.

- GH-260

Reply:
Hi Sebastian. Thanks for your review. I can only say that this spot has been found out by educated guessing and a bit more trial-and-error, so it was kind of empirically determined. Apologies that I can't come up with a better reasoning, but I can offer to submit another repro PR without all the other changes here, in order to better demonstrate the issue.